File: packages/contracts/src/building-safety.ts
import type { EntityId, VersionedRecord } from "./common";

export type BuildingSafetyItemType = "fire_extinguisher" | "dae" | "periodic_check";
export type BuildingSafetyItemStatus = "active" | "archived";
export type BuildingSafetyAlertStatus = "ok" | "due_soon" | "overdue" | "archived";
export type BuildingSafetyAlertType = "due_soon" | "overdue";

export interface BuildingSafetyItemRecord extends VersionedRecord {
  organization_id: EntityId;
  site_id: EntityId;
  site_name: string;
  item_type: BuildingSafetyItemType;
  name: string;
  next_due_date: string;
  last_checked_at: string | null;
  status: BuildingSafetyItemStatus;
  alert_status: BuildingSafetyAlertStatus;
  notes: string | null;
}

export interface BuildingSafetyAlertRecord {
  item_id: EntityId;
  site_id: EntityId;
  site_name: string;
  item_name: string;
  item_type: BuildingSafetyItemType;
  alert_type: BuildingSafetyAlertType;
  due_date: string;
  message: string;
}

export interface BuildingSafetyItemCreateRequest {
  site_id: EntityId;
  item_type: BuildingSafetyItemType;
  name: string;
  next_due_date: string;
  last_checked_at?: string | null;
  notes?: string | null;
}

export interface BuildingSafetyItemUpdateRequest {
  name?: string | null;
  next_due_date?: string | null;
  last_checked_at?: string | null;
  status?: BuildingSafetyItemStatus | null;
  notes?: string | null;
}

File: apps/api/app/schemas/building_safety.py
from datetime import date
from uuid import UUID

from pydantic import BaseModel, Field

from app.db.models.building_safety_item import BuildingSafetyItemStatus, BuildingSafetyItemType
from app.schemas.common import BaseReadModel


class BuildingSafetyItemRead(BaseReadModel):
    organization_id: UUID
    site_id: UUID
    site_name: str
    item_type: BuildingSafetyItemType
    name: str
    next_due_date: date
    last_checked_at: date | None
    status: BuildingSafetyItemStatus
    alert_status: str
    notes: str | None


class BuildingSafetyAlertRead(BaseModel):
    item_id: UUID
    site_id: UUID
    site_name: str
    item_name: str
    item_type: BuildingSafetyItemType
    alert_type: str
    due_date: date
    message: str


class BuildingSafetyItemCreateRequest(BaseModel):
    site_id: UUID
    item_type: BuildingSafetyItemType
    name: str = Field(min_length=2, max_length=160)
    next_due_date: date
    last_checked_at: date | None = None
    notes: str | None = Field(default=None, max_length=2000)


class BuildingSafetyItemUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=160)
    next_due_date: date | None = None
    last_checked_at: date | None = None
    status: BuildingSafetyItemStatus | None = None
    notes: str | None = Field(default=None, max_length=2000)
